packages/flutter/lib/src/widgets/autocomplete.dart

```diff
@@ -502,6 +505,30 @@ class _RawAutocompleteState<T extends Object> extends State<RawAutocomplete<T>>
     };
     _updateActions();
     _updateOverlay();
+
+    WidgetsBinding.instance.addObserver(this);
+    WidgetsBinding.instance.addPostFrameCallback((_) => _getFieldWidth());
+  }
+
+  bool _isOffstage = true;
+  double? _fieldWidth;
+  final GlobalKey _textFieldKey = GlobalKey();
+
+  void _getFieldWidth() {
+    assert(_textFieldKey.currentContext != null, "The fieldKey must be assigned to a TextFormField");
+    if (_textFieldKey.currentContext != null) {
```

Comment:
By using both this `if` and the `assert` above, you're saying that it should throw an error in development mode, but in production it should just silently not update the _fieldWidth and _isOffstage.  Is that what you intended?  Just making sure.

Reply:
yes

Reply:
If the key is not assigned to any widget and we are in production, _isOffstage will never be set to `false` and thus the Overlay will never be displayed (even with double.maxFinite as width constraints).
Should I move the `setState(() => _isOffstage = false)` to after the `if` to run it no matter what?

Reply:
Please ignore previous question, I removed Offstage as it was actually useless